snuba/datasets/__init__.py

```diff
@@ -114,6 +114,15 @@ def get_bulk_loader(self, source, dest_table):
     def get_query_schema(self):
         raise NotImplementedError('dataset does not support queries')
 
+    def get_prewhere_keys(self) -> List[str]:
```

Comment:
This doesn't need to strictly be a `List`, you just need a collection of items that can be iterated multiple times and has a stable order, so a [`Sequence`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence) makes sense here (and has the benefit of no expectation of mutability):

```suggestion
    def get_prewhere_keys(self) -> Sequence[str]:
```

(You'll also need to import `Sequence` from typing instead.)

Reply:
Oh cool. Good to know!